### DISCUSSION TITLE
"Python int too large to convert to C long"

### DISCUSSION THREAD

**Discussion by DChu5** [Project-MONAI/MONAI#8131] [Q&A]
Hi there. I'm currently learning to use the classification tutorial at https://github.com/Project-MONAI/tutorials/blob/main/3d_classification/densenet_training_array.ipynb with the hopes of training a model based on a collection of CT scans. My main constraint is that my workplace will not allow for installation of numpy version <2 to fulfill the requirements for MONAI, citing security concerns. I am trying to make the most of the situation and potentially find a work-around in the meantime but I am running into the error "Python int too large to convert to C long" when defining the image transforms "train_transforms" and "val_transforms".

I would greatly appreciate any and all help that you could provide. Thank you!

**Comment by KumoLiu** [comment]
Hi @DChu5, thanks for your interest here.

It seems that the issue you're facing isn't directly related to MONAI but rather to the data type exceeding the limits for the system's C long type. The error “Python int too large to convert to C long” typically occurs when you're passing an integer to a C function or library that exceeds the range of long on your system (either 32-bit or 64-bit). 
You could check the data type of your inputs or maybe convert large integers to smaller data types.

Hope it helps, thanks.

**Reply by DChu5** [reply]
Thanks a bunch for your reply! The error occurs at "train_transforms = Compose([ScaleIntensity(), EnsureChannelFirst(), Resize((96, 96, 96)), RandRotate90()])" and the reason I suspected it to be related to Numpy is that if I downgrade on my personal computer to version <2, the code runs without issues. I'll keep working with it but would appreciate any further guidance.
